### PR TITLE
Update examples (1)

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: docs/examples/micropayment.rst [hunk 1]
```diff
@@ -94,12 +109,26 @@ Here is a JavaScript function that creates the proper signature for the ``Receiv
...
 Extracting the Signature Parameters
 -----------------------------------
 
-Signatures produced by web3.js are the concatenation of ``r``, ``s`` and ``v``, so the first step is to split these parameters apart. You can do this on the client-side, but doing it inside the smart contract means you only need to send one signature parameter rather than three. Splitting apart a byte array into component parts is a mess, so we use `inline assembly <assembly>`_ to do the job in the ``splitSignature`` function (the third function in the full contract at the end of this section).
+Signatures produced by web3.js are the concatenation of ``r``,
+``s`` and ``v``, so the first step is to split these parameters
+apart. You can do this on the client-side, but doing it inside
+the smart contract means you only need to send one signature
+parameter rather than three. Splitting apart a byte array into
+component parts is a mess, so we use
```
Comment (bshastry) [inline docs/examples/micropayment.rst]:
```suggestion
its constituent parts is a mess, so we use
```

File: docs/examples/micropayment.rst [hunk 2]
```diff
@@ -180,26 +209,44 @@ The full contract
...
     2. Alice signs messages that specify how much of that Ether is owed to the recipient. This step is repeated for each payment.
     3. Bob "closes" the payment channel, withdrawing their portion of the Ether and sending the remainder back to the sender.
 
 .. note::
-  Only steps 1 and 3 require Ethereum transactions, step 2 means that the sender transmits a cryptographically signed message to the recipient via off chain methods (e.g. email). This means only two transactions are required to support any number of transfers.
-
-Bob is guaranteed to receive their funds because the smart contract escrows the Ether and honours a valid signed message. The smart contract also enforces a timeout, so Alice is guaranteed to eventually recover their funds even if the recipient refuses to close the channel. It is up to the participants in a payment channel to decide how long to keep it open. For a short-lived transaction, such as paying an internet café for each minute of network access, or for a longer relationship, such as paying an employee an hourly wage, a payment could last for months or years.
+  Only steps 1 and 3 require Ethereum transactions, step 2 means that the sender
```
Comment (bshastry) [inline docs/examples/micropayment.rst:229]: Is the extra indentation in this and the next few lines intended?

File: docs/examples/micropayment.rst [hunk 3]
```diff
@@ -180,26 +209,44 @@ The full contract
...
+  any number of transfers.
+
+Bob is guaranteed to receive their funds because the smart contract escrows the
+Ether and honours a valid signed message. The smart contract also enforces a
+timeout, so Alice is guaranteed to eventually recover their funds even if the
+recipient refuses to close the channel. It is up to the participants in a payment
+channel to decide how long to keep it open. For a short-lived transaction,
+such as paying an internet café for each minute of network access, or for a
+longer relationship, such as paying an employee an hourly wage, a payment 
+ould last for months or years.
```
Comment (bshastry) [inline docs/examples/micropayment.rst]:
```suggestion
could last for months or years.
```

**Comment by leonardoalt** [comment]
Style failing

File: docs/examples/micropayment.rst [hunk 4]
```diff
@@ -180,26 +209,44 @@ The full contract
...
+  transmits a cryptographically signed message to the recipient via off chain
+  methods (e.g. email). This means only two transactions are required to support
+  any number of transfers.
+
+Bob is guaranteed to receive their funds because the smart contract escrows the
+Ether and honours a valid signed message. The smart contract also enforces a
+timeout, so Alice is guaranteed to eventually recover their funds even if the
+recipient refuses to close the channel. It is up to the participants in a payment
+channel to decide how long to keep it open. For a short-lived transaction,
+such as paying an internet café for each minute of network access, or for a
```
Comment (bshastry) [inline docs/examples/micropayment.rst:239]: I would suggest rephrasing this sentence as follows or something like that: `such as paying an internet café for each minute of network access, the payment channel may be kept open for a limited duration. On the other hand, for a recurring payment, such as paying an employee an hourly wage, the payment could be kept open for several months or years.`

File: docs/examples/micropayment.rst [hunk 5]
```diff
@@ -218,7 +265,8 @@ Because of this, only one of the messages sent is redeemed. This is why
 each message specifies a cumulative total amount of Ether owed, rather than the
 amount of the individual micropayment. The recipient will naturally choose to
 redeem the most recent message because that is the one with the highest total.
-The nonce per-message is not needed anymore, because the smart contract only honors a single message. The address of the smart contract is still used
+The nonce per-message is not needed anymore, because the smart contract only
+honors a single message. The address of the smart contract is still used
```
Comment (bshastry) [inline docs/examples/micropayment.rst]:
```suggestion
honours a single message. The address of the smart contract is still used
```

**Review by bshastry — CHANGES_REQUESTED** [review]

File: docs/examples/micropayment.rst [hunk 6]
```diff
@@ -180,26 +209,44 @@ The full contract
...
+  methods (e.g. email). This means only two transactions are required to support
+  any number of transfers.
+
+Bob is guaranteed to receive their funds because the smart contract escrows the
+Ether and honours a valid signed message. The smart contract also enforces a
+timeout, so Alice is guaranteed to eventually recover their funds even if the
+recipient refuses to close the channel. It is up to the participants in a payment
+channel to decide how long to keep it open. For a short-lived transaction,
+such as paying an internet café for each minute of network access, or for a
+longer relationship, such as paying an employee an hourly wage, a payment
```
Comment (bshastry) [inline docs/examples/micropayment.rst]:
```suggestion
longer relationship, such as paying an employee an hourly wage, a payment
```

**Comment by stackenbotten** [comment]
There was an error when running `chk_coding_style` for commit `b01d899d56e09ae718eca4e962cf2344d8c842f9`:
```
Error: Trailing whitespace found:
 docs/examples/micropayment.rst:240:longer relationship, such as paying an employee an hourly wage, a payment 

```
Please check that your changes are working as intended.

File: docs/examples/micropayment.rst (hunk 2 above)
Comment (chriseth) [reply docs/examples/micropayment.rst:229]: Should be the same indentation as the non-broken line was before.

File: docs/examples/micropayment.rst (hunk 4 above)
Comment (chriseth) [reply docs/examples/micropayment.rst:239]: I just added line breaks. Can you make a PR after this is merged?

Comment (bshastry) [reply docs/examples/micropayment.rst:239]: Okay (CC https://github.com/ethereum/solidity/issues/7990)

File: docs/examples/micropayment.rst (hunk 2 above)
Comment (bshastry) [reply docs/examples/micropayment.rst:229]: The second line onwards are left aligned in the original text but not in this PR, is that also intended?

**Review by bshastry — DISMISSED** [review]

**Pull request merged**